Makefile

```diff
@@ -1,42 +1,113 @@
+BPFTOOL ?= bpftool
+CARGO ?= cargo
+CLANG ?= clang
 CRUNTIME ?= docker
+CURL ?= curl
 KERNEL_TAG ?=
 EXTRA_DOCKER_FLAGS ?=
+CONTAINERIZED_BUILD ?= 1
+
+PREFIX ?= $(DESTDIR)/usr/local
+BINDIR ?= $(PREFIX)/bin
+UNITDIR ?= $(PREFIX)/lib/systemd/system
 
 ifneq ($(KERNEL_TAG),)
 EXTRA_DOCKER_FLAGS += --build-arg KERNEL_TAG=$(KERNEL_TAG)
 endif
 
+ifeq ($(CONTAINERIZED_BUILD),0)
+OUTDIR = target/release
+else
+OUTDIR = out
+endif
+
 .PHONY: all
-all: build fmt lint
+all: gen build fmt lint
+
+.PHONY: gen
+gen:
+ifeq ($(CONTAINERIZED_BUILD),0)
+	# Try to generate vmlinux.h. If not possible, download the newest one
+	# from the libbpf community.
+	$(BPFTOOL) btf dump file \
+		/sys/kernel/btf/vmlinux format c > \
+		src/bpf/vmlinux.h || \
+		$(CURL) -L https://raw.githubusercontent.com/libbpf/libbpf-bootstrap/master/vmlinux/vmlinux_508.h \
+		--output src/bpf/vmlinux.h
+	$(CARGO) libbpf build --clang-path $(CLANG)
+	$(CARGO) libbpf gen
+else
+	$(CRUNTIME) build \
+		--build-arg USER_ID=$(shell id -u) \
+		--build-arg GROUP_ID=$(shell id -g) \
+		--target gen \
+		--tag enclave-gen \
+		$(EXTRA_DOCKER_FLAGS) \
+		.
+	$(CRUNTIME) run \
+		--rm -i \
+		--user "$(shell id -u):$(shell id -g)" \
+		-v $(shell pwd):/usr/local/src/enclave \
+		enclave-gen
+endif
 
 .PHONY: build
-build:
```

Comment:
You shouldn't remove the build target, or?

Reply:
Yes, good catch

Reply:
fixed